src/Compilers/CSharp/Portable/Symbols/Source/ParameterHelpers.cs

```diff
@@ -648,7 +651,7 @@ static void addERR_ParamsCantBeWithModifier(BindingDiagnosticBag diagnostics, Sy
                 diagnostics.Add(ErrorCode.ERR_MethodArgCantBeRefAny, parameterSyntax.Location, parameter.Type);
```

Comment:
Is it possible for us to avoid using `parameter.Type` here as well?

Reply:
I think we're ok here because we've already checked that `parameter.TypeWithAnnotations` is a restricted type. That means the type is not a type parameter which is the only case where we'd have an unresolved type.

Reply:
Will this break when we add `allow T : ref struct`? No need to adjust the implementation based on something we haven't done yet, but would be good to have a sense of what our path forward would be in that case.

Reply:
> Will this break when we add `allow T : ref struct`?

Yes, this will break when we support `ref struct` types as type arguments.